tests/test_samples.py

```diff
@@ -573,6 +573,27 @@ def test_limit_assignment():
     assert ns.limits['nlive'][1] == 125
 
 
+def test_xmin_xmax_1d():
+    """Test to provide a solution to #89"""
+    np.random.seed(3)
+    ns = NestedSamples(root='./tests/example_data/pc')
+    fig, ax = ns.plot_1d('x0', plot_type='hist')
+    assert ax['x0'].get_xlim() != (-1, 1)
+    fig, ax = ns.plot_1d('x0', plot_type='hist', xmin=-1, xmax=1)
+    assert ax['x0'].get_xlim() == (-1, 1)
+
+
+def test_equal_min_max():
+    """Test to provide a solution to #89"""
+    np.random.seed(3)
+    ns = NestedSamples(root='./tests/example_data/pc')
+    with pytest.raises(ValueError):
+        ns.plot_2d(['x0', 'x1', 'x2'], xmin=3, xmax=3)
+
+    ns.limits['x0'] = (3, 3)
+    ns.plot_2d(['x0', 'x1', 'x2'])
```

Comment:
The code snippet:
```
ns = NestedSamples(root="./tests/example_data/pc")
ns.limits['x0'] = (0, 5)
ns.plot_2d(['x0', 'x1', 'x2']);
```
does not produce the expected output:

![equal_min_max](https://user-images.githubusercontent.com/18258042/85470621-344f4e80-b5a7-11ea-8a98-ce86e561458c.png)

Reply:
This actually already is an issue in `master`.

Reply:
This is an unexpected side-effect. I don't think I've tried to set xmin and xmax less than the actual ranges before this. What's happening is that `ncompress` is applied to all the samples, and then the xmin and xmax limits are applied to these compressed samples, so there aren't enough points to do a decent contour plot.

This is not an easy fix, as at the moment it will require separate treatment for `'fastkde'`, `'scatter'` and `'kde'`, and ideally these would be treated in the same way. This also likely interacts with #99 and #95. 

Given that this affects master as well, and is likely better addressed in conjunction with the fixes to #99 and #95, could you raise this as a separate issue, or add it to one of the others (with your nice MWE copy-pasted)?

Reply:
This crossed with a comment from me in #99, where I wonder whether we should have separate `xmin` and `xmax` for compression and plotting.

Reply:
> Given that this affects master as well, and is likely better addressed in conjunction with the fixes to #99 and #95, could you raise this as a separate issue, or add it to one of the others (with your nice MWE copy-pasted)?

Will do.